Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -284,6 +292,39 @@ public final class SwiftTargetBuildDescription {
                 self.resourceBundleInfoPlistPath = infoPlistPath
             }
         }
+
+        try self.generateResourceEmbeddingCode()
+    }
+
+    // FIXME: This will not work well for large files, as we will store the entire contents, plus its byte array representation in memory and also `writeIfChanged()` will read the entire generated file again.
+    private func generateResourceEmbeddingCode() throws {
+        guard needsResourceEmbedding else { return }
+
+        let stream = BufferedOutputByteStream()
+        stream <<< """
+        \(self.toolsVersion < .vNext ? "import" : "@_implementationOnly import") struct Foundation.Data
+
+        struct PackageResources {
+
+        """
+
+        try resources.forEach {
+            guard $0.rule == .embed else { return }
+
+            let variableName = $0.path.basename.spm_mangledToC99ExtendedIdentifier()
+            let fileContent = try Data(contentsOf: URL(fileURLWithPath: $0.path.pathString)).map { String($0) }.joined(separator: ",")
+
+            stream <<< "static let \(variableName) = Data([\(fileContent)])\n"
```

Comment:
would be nice to just do byte array instead of data since not all users would want to use foundation